python/plugins/db_manager/db_plugins/spatialite/connector.py

```diff
@@ -44,7 +45,13 @@ def __init__(self, uri):
 
         try:
             self.connection = sqlite.connect(self._connectionInfo())
-
+            self.connection.enable_load_extension(True)
+            c = self._get_cursor()
+            if os.name == 'nt':
```

Comment:
Is this correct? Is 'nt' not Windows and that's where the .dll should be used?

Would it make sense to put this init code into a helper module (e.g. qgis.spatialite), what do you think?